drizzlepac/generate_final_product_filenames.py

```diff
@@ -1,9 +1,188 @@
+sep_str = 'single exposure product {:02d}'
+fp_str = 'filter product {:02d}'
+tdp_str = 'total detection product {:02d}'
+
 """This script is development space for code that will be most likely
     absorbed into a larger script later on.
 
 """
+def interpret_input(results, exposure_limit=2.0):
+    """
+    Interpret the database query for a given visit to prepare the returned
+    values for use in generating the names of all the expected output products.
+    
+    Input will have format of:
+        ia1s70jtq_flt.fits,11150,A1S,70,70,149.232269,F110W,IR,/ifs/archive/ops/hst/public/ia1s/ia1s70jtq/ia1s70jtq_flt.fits
+        ia1s70iwq_flt.fits,11150,A1S,70,70,0.91161000000000003,F160W,IR,/ifs/archive/ops/hst/public/ia1s/ia1s70iwq/ia1s70iwq_flt.fits
+        
+    Output dict will have format (as needed by further code for creating the
+        product filenames) of:
+    
+        obs_info_dict["single exposure product 00"] = {'info': '11150 A1S WFC3 IR F110W IA1S70JTQ', 
+                                                        'files':['ia1s70jtq_flt.fits']}
+        obs_info_dict["single exposure product 01"] = {'info': '11150 A1S WFC3 IR F160W IA1S70JWQ', 
+                                                        'files':['ia1s70jwq_flt.fits']}
+                                                        
+        obs_info_dict["filter product 00"] = {"info": '11150 A1S WFC3 IR F110W', 
+                                              "files":['ia1s70jtq_flt.fits']}
+        obs_info_dict["filter product 01"] = {"info": '11150 A1S WFC3 IR F160W', 
+                                              "files":['ia1s70jwq_flt.fits']}
+        obs_info_dict["total detection product 00"] = {'info': '11150 A1S WFC3 IR',
+                                                       'files':['ia1s70jtq_flt.fits',
+                                                                'ia1s70iwq_flt.fits']}
+    
+    """
+    colnames = ['filename', 'proposal_id', 'program_id', 'obset_id', 'visit_id',
+                'exptime', 'filters', 'detector', 'pathname']
+    instrument_dict = {'i':'WFC3', 'j':'ACS','o':'STIS','u':'WFPC2','x':'FOC','w':'WFPC'}
+    visit_table = Table.read(results, format='ascii.fast_no_header')
+    # label each column with a descriptive name
+    for col, cname in zip(visit_table.colnames, colnames):
+        visit_table[col].name = cname
+    # Add INSTRUMENT column
+    instr= instrument_dict[visit_table['filename'][0][0]]
+    # convert input to an Astropy Table for parsing
+    visit_table.add_column(Column([instr]*len(visit_table)),name='instrument')
+    # parse Table into a tree-like dict
+    visit_tree = build_visit_tree(visit_table, exposure_limit=exposure_limit)
+    # Now create final dict
+    visit_dict = parse_visit_tree(visit_tree)
+    
+    return visit_dict
+
+# Translate the database query on a visit into actionable lists of filenames
+def build_visit_tree(visit_table, exposure_limit=2.0):
```

Comment:
I am not sure about this whole function. I have a feeling it can be written in a different way. If anything I will comment on the main "thread".